teamNewDev/jobhunt
# #05 Setup ESLint

#### What does this PR do?

Integrates ESLint with Airbnb JS style guide for maintaining code quality and ensuring consistency across the team

#### Description of Task to be completed?

- initialize package.json
- install ESLint as a dev dependency
- install the Airbnb JS style guide config with its peer dependencies
- create a `.eslintrc.json` file for lint configuration
- update rules in `.eslintrc.json` to ignore linebreaks formatting
- update rules in `.eslintrc.json` to enforce single quotes with severity of 1 (warn)
- update rules in `.eslintrc.json` to enforce no unnecessary semicolons with severity of 1 (warn)
- add lint script in `package.json`

#### How should this be manually tested?

1. clone the repository
2. switch to `workflow/05/setup-eslint` branch
3. upon running `npm install` in the command line, all project dependencies should be installed in the local dev environment
4. upon running `npm run lint`, ESLint should run and report any linting errors based on the rules set in the `.eslintrc.json` file
  - the following error should be displayed in the console:
   > warning  Unexpected console statement  no-console

  the error is due to the `console.log` currently in the `index,js`. The linter has been configured to throw a warning on each `console.log` in the JS file.

#### Any background context you want to provide?

Integrating ESLint into the project is crucial for maintaining code quality, ensuring consistency across the team, catching errors early, and enforcing JavaScript best practices. Its integration into the build process further ensures that only quality code gets deployed by throwing an error when the code does not meet the standard.

#### What are the relevant GitHub Issues?

#5 

#### Screenshots (if appropriate)


![new rules - eslint](https://github.com/teamNewDev/jobhunt/assets/42160985/62086a5a-bb95-4324-bc0c-a6a637868579)


#### Questions:

None
